msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/ltw/TestCase2582296.java

```diff
@@ -0,0 +1,125 @@
+import com.microsoft.identity.client.ui.automation.broker.BrokerLTW;
+import com.microsoft.identity.client.ui.automation.broker.BrokerMicrosoftAuthenticator;
+import com.microsoft.identity.labapi.utilities.client.LabQuery;
+import com.microsoft.identity.labapi.utilities.constants.TempUserType;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Map;
+
+// Test Case 2582296: If LTW is the active broker, and request is made through Authenticator from Multiple WorkplaceJoin API from a legacy broker test app, nothing should break.
+//https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2582296
+@LTWTests
+@SupportedBrokers(brokers = {BrokerLTW.class})
+public class TestCase2582296 extends AbstractMsalBrokerTest {
+    @Test
+    public void test_2582296() {
+        final String username = mLabAccount.getUsername();
+        final String password = mLabAccount.getPassword();
+        final String tenantId = mLabAccount.getHomeTenantId();
+
+        // Install new Authenticator with broker SDK changes of broker selection logic
+        final BrokerMicrosoftAuthenticator brokerMicrosoftAuthenticator = new BrokerMicrosoftAuthenticator();
+        brokerMicrosoftAuthenticator.install();
+
+        // Install old BrokerHost app with no broker SDK changes of broker selection logic
+        final BrokerHost brokerHost = new BrokerHost(BrokerHost.BROKER_HOST_WITHOUT_BROKER_SELECTION_APK);
+        brokerHost.install();
+        brokerHost.launch();
+
+        // Switch to Multiple WPJ API from hamburger menu on the top left
+        // Enter username retrieved
+        // Click on "Device Registration" button
+        brokerHost.multipleWpjApiFragment.performDeviceRegistration(username, password);
+
+        // Enter tenantId and Click on "Get Blob" button
+        final String blob = brokerHost.multipleWpjApiFragment.getBlob(tenantId);
+        Assert.assertTrue(!TextUtils.isEmpty(blob));
+
+        // Click on "Get all records" button
+        // A popup with all the registered entries is shown.
+        final List<Map<String, String>> records = brokerHost.multipleWpjApiFragment.getAllRecords();
+        Assert.assertEquals(1, records.size());
+        final Map<String, String> record = records.get(0);
+        Assert.assertEquals(username, record.get("Upn"));
+        Assert.assertEquals(tenantId, record.get("TenantId"));
+
+        // Click on "Get record by Upn" button
+        final Map<String, String> recordByUpn = brokerHost.multipleWpjApiFragment.getRecordByUpn(username);
+        Assert.assertEquals(record, recordByUpn);
+
+        // Click on "Get State" button
+        final String state = brokerHost.multipleWpjApiFragment.getDeviceState(username);
+        Assert.assertTrue(state.contains("DEVICE_VALID"));
+
+        // Click on "Install certificate" button
+        brokerHost.multipleWpjApiFragment.installCertificate(username);
+
+        // Click on "Get device token" button
```

Comment:
Seems we're missing step 15 from https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2582296, is this intentional

Reply:
Steps 14, 15 and 16 are all in this `installCertificate` method.